src/legacy/core_plugins/kbn_vislib_vis_types/public/editors/metrics_axes_options.tsx

```diff
@@ -0,0 +1,308 @@
+        const lastSeriesAggField = get(lastSeriesAgg, `${matchingSeries[0].id}.field`);
+        const matchingSeriesAggType = get(matchingSeries, '[0]type.name', '');
+        const matchingSeriesAggField = get(matchingSeries, '[0]params.field.name', '');
+
+        const aggTypeIsChanged = lastSeriesAggType !== matchingSeriesAggType;
+        const aggFieldIsChanged = lastSeriesAggField !== matchingSeriesAggField;
+
+        lastMatchingSeriesAgg[matchingSeries[0].id] = {
+          type: matchingSeriesAggType,
+          field: matchingSeriesAggField,
+        };
+        lastLabels[axis.id] = newCustomLabel;
+
+        if (
+          aggTypeIsChanged ||
+          aggFieldIsChanged ||
+          axis.title.text === '' ||
+          lastCustomLabels[axis.id] === axis.title.text
+        ) {
+          // Override axis title with new custom label
+          axes[axisNumber] = { ...axes[axisNumber], title: { ...axis, text: newCustomLabel } };
+          isAxesChanged = true;
+        }
+      }
+    });
+
+    if (isAxesChanged) {
+      setValue('valueAxes', axes);
+    }
+
+    setLastSeriesAgg(lastMatchingSeriesAgg);
+    setLastCustomLabels(lastLabels);
+  }, [stateParams.valueAxes, stateParams.seriesParams, setValue]);
+
+  const getUpdatedAxisName = useCallback(
+    (axisPosition: ValueAxis['position']) => {
+      const axisName = capitalize(axisPosition) + AXIS_PREFIX;
+      const lastAxisNameNumber = axesNumbers[axisPosition];
+      let nextAxisNameNumber;
+
+      if (!lastAxisNameNumber) {
+        nextAxisNameNumber = stateParams.valueAxes.reduce(countNextAxisNumber(axisName, 'name'), 1);
+      } else {
+        nextAxisNameNumber = lastAxisNameNumber + 1;
+      }
+
+      setAxesNumbers({ ...axesNumbers, [axisPosition]: nextAxisNameNumber });
+
+      return `${axisName}${nextAxisNameNumber}`;
+    },
+    [stateParams.valueAxes, axesNumbers]
+  );
+
+  const onValueAxisPositionChanged = useCallback(
+    (index: number, value: ValueAxis['position']) => {
+      const valueAxes = [...stateParams.valueAxes];
+
+      valueAxes[index] = {
+        ...valueAxes[index],
+        name: getUpdatedAxisName(value),
+        position: value,
+      };
+      setValue('valueAxes', valueAxes);
+    },
+    [stateParams.valueAxes, getUpdatedAxisName]
+  );
+
+  const onCategoryAxisPositionChanged = useCallback(
+    (chartPosition: Axis['position']) => {
+      const isChartHorizontal = isAxisHorizontal(chartPosition);
+      if (isChartHorizontal !== isCategoryAxisHorizontal) {
+        setIsCategoryAxisHorizontal(isChartHorizontal);
+      }
+
+      stateParams.valueAxes.forEach((axis, index) => {
+        if (isAxisHorizontal(axis.position) === isChartHorizontal) {
+          const position = mapPosition(axis.position);
+          onValueAxisPositionChanged(index, position);
+        }
+      });
+    },
+    [stateParams.valueAxes, isCategoryAxisHorizontal, onValueAxisPositionChanged]
```

Comment:
Then we would get rid of:

```suggestion
    [stateParams.valueAxes, onValueAxisPositionChanged]
```